py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -95,6 +95,8 @@ def __call__(self, ignored):
 
 
 def _element_if_visible(element, visibility=True):
+    if isinstance(element, str) or isinstance(element, dict):
+           raise StaleElementReferenceException("Invalid locator")
```

Comment:
I'm sorry, but this pull request will most likely won't be accepted because it doesn't solve the problem for everyone. I suspect that you may have a bug in your code where the `elem` is an actual element and not a `str`ing that represents a CSS locator. If there is a Dict or String being produced by Firefox and Chrome you will have to file a bug report, since WebDriver is only suppose to produce WebElements. The only exception to this is if you extract data via a javascript call (`execute_script`).

You should not ignore exceptional cases too, that is most likely a bug in your code.